### PR TITLE
Feature/241/add docs

### PR DESCRIPTION
**Pull request by tnobody**

### CODE REVIEW

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 1]
```diff
@@ -1,28 +1,369 @@
...
+     * <div>Hello <i>World</i></div>
+     * <div id="pending-request">
+     *     There are 5 pending request
+     * </div>
+     * ```
+     * ```typescript
+     * const containsIpsumDolor = await _containsText(_paragraph('intro'), 'ipsum dolor') // -> true
+     * const containsYpsumDolor = await _containsText(_paragraph('intro'), 'ypsum dolor') // -> false
+     *
+     * const containsText = await _containsText(_div(0), 'Hello World');
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]: The comment with the return value is missing here, isn't it? Same below

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 2]
```diff
@@ -1,28 +1,369 @@
...
+     * await _containsHTML(_div(0), 'Non existent'); // -> false
+     * ```
+     *
+     * @param query
+     * @param html
+     */
     _containsHTML(query: SahiElementQueryOrWebElement, html: string): Promise<boolean>;
+
+    /**
+     * Check whether an _parent_ element contains a _child_ element.
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
     * Checks if a _parent_ element contains a _child_ element.
```

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 3]
```diff
@@ -1,28 +1,369 @@
...
+     * ```html
+     * <div id="parent">
+     *     <div id="child-a"></div>
+     *     <div id="child-b"></div>
+     * </div>
+     * ```
+     *
+     * ```typescript
+     * await _contains(_div('parent'), _div('child-b')) // -> true
+     * await _contains(_div('child-a'), _div('child-b')) // -> true
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
     * await _contains(_div('child-a'), _div('child-b')) // -> false
```

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 4]
```diff
@@ -1,28 +1,369 @@
...
+     * ```
+     *
+     * @param parent
+     * @param child
+     */
     _contains(parent: SahiElementQueryOrWebElement, child: SahiElementQueryOrWebElement): Promise<boolean>
+
+    /**
+     * Retrieves the current text-content of the documents [`title`]{@link https://developer.mozilla.org/de/docs/Web/HTML/Element/title} element.
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
     * Retrieves the current text content of the documents [`title`]{@link https://developer.mozilla.org/de/docs/Web/HTML/Element/title} element.
```

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 5]
```diff
@@ -1,28 +1,369 @@
...
+    * _style returns the computed style that is finally applicable to the element.
+    * If the element inherits the named style from
+    * its parent, the parent will be queried for its value.  Where possible,
+    * color values will be converted to their hex representation (e.g. #00ff00
+    * instead of rgb(0, 255, 0)).
+    *
+    * Accessing style directly as an attribute will not give computed style.
+    * Always use _style instead.
+    *
+    * _Warning:_ the value returned will be as the browser interprets it, so
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
    * _Warning:_ A proper assertion will be difficult as some browser may interpret it differently.
```

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 6]
```diff
@@ -1,28 +1,369 @@
...
+
+    /**
+    * Styles in HTML elements are calculated by the browser based on various CSS rules.
+    * _style returns the computed style that is finally applicable to the element.
+    * If the element inherits the named style from
+    * its parent, the parent will be queried for its value.  Where possible,
+    * color values will be converted to their hex representation (e.g. #00ff00
+    * instead of rgb(0, 255, 0)).
+    *
+    * Accessing style directly as an attribute will not give computed style.
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
    * Accessing the style directly as an attribute will not return a computed style.
```

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 7]
```diff
@@ -1,28 +1,369 @@
...
+    */
     _style(query: SahiElementQueryOrWebElement, attr: string): Promise<string>;
+
+    /**
+     * Returns a tuple describing an element's location, in pixels relative to
+     * the document element.
+     *
+     * @example
+     * ```typescript
+     * // Assuming an modal overlay element at x: 300 and y: 400
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
     * // Assuming a modal overlay element at x: 300 and y: 400
```

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 8]
```diff
@@ -1,28 +1,369 @@
...
+    * @param query
+    * @param attr
+    */
     _style(query: SahiElementQueryOrWebElement, attr: string): Promise<string>;
+
+    /**
+     * Returns a tuple describing an element's location, in pixels relative to
+     * the document element.
+     *
+     * @example
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]: The formating doesn't look quite right as the `example` is in the code box.
example

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 9]
```diff
@@ -1,28 +1,369 @@
...
+     * // Assuming an modal overlay element at x: 300 and y: 400
+     * const [x,y] = await _position(_div('modal')) // -> [300, 400];
+     * ```
+     *
+     * @param query
+     */
     _position(query: SahiElementQueryOrWebElement): Promise<[number, number]>;
+
+    /**
+     * Retrieves the selection from a user in the current document as string.
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
     * Retrieves the selection from a user in the current document as a string.
```

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 10]
```diff
@@ -1,28 +1,369 @@
...
+     * This is most likely the text what will set to the clipboard if a user
+     * copies the selection
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
     * copies the selection.
```

File: packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts [hunk 11]
```diff
@@ -1,28 +1,369 @@
...
+     */
     _getSelectionText(): Promise<string>
+
+    /**
+     * Executes a the given query and returns a "native" [`WebElement`]{@link https://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebElement.html} instance
```
Comment (cliffle) [inline packages/sakuli-legacy/src/context/sahi/fetch/fetch-api.interface.ts]:
```suggestion
     * Executes the given query and returns a "native" [`WebElement`]{@link https://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebElement.html} instance
```

**Comment by tnobody** [comment]
@cliffle, sorry somehow I didn't saw the remaining suggestions from your review. With the last commit everything should be considered.

**Pull request merged**